File: dotnet/Solid/Solid/LSP/Example2/Domain/WorldPayPayment.cs
using Solid.LSP.Example2.External;

namespace Solid.LSP.Example2.Domain
{
    public class WorldPayPayment : Payment
    {
        public string Username { private get; set; }
        public string Password { private get; set; }
        public string ProductId { private get; set; }

        public override string Refund(decimal amount, string transactionId)
        {
            WorldPayWebService worldPayWebService = new WorldPayWebService();
            string response = worldPayWebService.MakeRefund(amount, transactionId, Username, Password, ProductId);
            return response;
        }
    }
}

File: java/liskov-plus-start/src/main/java/no/bouvet/solid/lsp/domain/WorldPayPayment.java
package no.bouvet.solid.lsp.domain;

import java.math.BigDecimal;

import no.bouvet.solid.lsp.external.WorldPayWebService;

public class WorldPayPayment extends Payment {

    private String username;
    private String password;
    private String productId;

    public void setUsername(String username) {
        this.username = username;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public void setProductId(String productId) {
        this.productId = productId;
    }

    @Override
    public String refund(BigDecimal amount, String transactionId) {
        WorldPayWebService worldPayWebService = new WorldPayWebService();
        String response = worldPayWebService.makeRefund(amount, transactionId, username, password, productId);
        return response;
    }

}
